### ISSUE TITLE
Unexpected type for Rez submodule access as dictionary

### ISSUE THREAD

**Issue by m4tiz5zktonj** [markw65/monkeyc-optimizer#51]
```monkey-c
import Toybox.Lang;

(:typecheck(false))
function test1(index as Number) as ResourceId {
    var resourceSymbols = [:AppName];

    return Rez.Strings[resourceSymbols[index]] as ResourceId;
}

function test2(index as Number) as ResourceId {
    var resourceSymbols = [:AppName];

    // Expected $.test2 to return Toybox.Lang.ResourceId but got Any
    return (Rez.Strings as Rez.Strings)[resourceSymbols[index]];
}
```

Both ways above of accessing variables from `Rez` submodule are valid for Garmin's compiler, and PMC doesn't like the last one. Though `Rez.Strings as Rez.Strings` expression is ridiculous, it allows you to access resources without triggering typecheck warnings.

**Comment by markw65** [comment]
That's an interesting find. Garmin's type checker generally doesn't treat modules as being part of the type system, but here, it's allowing `Rez.Strings` as a type name, and amazingly it then deduces the correct type.

My type checker does treat modules as types, and so `(Rez.Strings as Rez.Strings)` is a no-op. But if you index into a module using a non-literal, I've always treated the result as `Any` (which is why you need the cast in `test1`).

But of course, in some cases, a module will only contain a particular type, or some small set of types. So I've fixed it to look through the actual definitions, and return the union (so in the case of Rez.Strings, it will return ResourceId).

So now neither version needs an `as ResourceId` cast on the result of the lookup. I'll try to get a release out today.

**Comment by markw65** [comment]
Fixed in v1.1.81 of the optimizer module, and 2.0.111 of the extension.
